arch/risc-v/Kconfig

```diff
@@ -188,6 +189,30 @@ config ARCH_MMU_TYPE_SV39
 	bool
 	default n
 
+config ARCH_HAVE_S_MODE
+	bool
+	default n
+
+if ARCH_HAVE_S_MODE
+
+# Option to run NuttX in supervisor mode. This is obviously not usable in
+# flat mode, is questionable in protected mode, but is mandatory in kernel
+# mode.
+#
+# Kernel mode requires this as M-mode uses flat addressing and the kernel
+# memory must be mapped in order to share memory between the kernel and
+# different user tasks which reside in virtual memory.
+
+config ARCH_USE_S_MODE
+	bool "Run the NuttX kernel in S-mode instead of M-mode"
```

Comment:
Change to "Run the NuttX kernel in S-mode" ?
Since "M" means machine, it's from risc-v , but there are other target support supervisor also, we shold make the description more general.

Reply:
Yes